### PR TITLE
Add {M}IBC to glossary

### PR DESCRIPTION
**Pull request by am11**
The terms IBC / MIBC are related to PGO and used in optimization related topics.

### CODE REVIEW

File: docs/project/glossary.md [hunk 1]
```diff
@@ -24,12 +24,14 @@ terminology.
...
+| IBC | Instrumented Block Counts - used as extension (`*.ibc`) for old PGO files. |
 | IPC | Inter-Process Communication. |
 | IL | Intermediate Language. Equivalent to CIL, also equivalent to [MSIL](https://docs.microsoft.com/dotnet/standard/managed-execution-process#compiling-to-msil). |
 | JIT | [Just-in-Time](https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/jit/ryujit-overview.md) compiler. RyuJIT is the code name for the next generation Just-in-Time(aka "JIT") for the .NET runtime. |
 | LCG | Lightweight Code Generation. An early name for [dynamic methods](https://github.com/dotnet/runtime/blob/main/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/DynamicMethod.cs). |
 | MD | MetaData. |
 | MDA | Managed Debugging Assistant - see [details](https://docs.microsoft.com/dotnet/framework/debug-trace-profile/diagnosing-errors-with-managed-debugging-assistants) (Note: Not in .NET Core, equivalent diagnostic functionality is made available on a case-by-case basis, e.g. [#9418](https://github.com/dotnet/runtime/issues/9418)) |
+| MIBC | Modern Instrumented Block Counts - used as extension (`*.mibc`) for modern PGO files. |
```
Comment (am11) [inline docs/project/glossary.md:34]: I am not sure if `M` stands for `Modern`, `Managed` or `Microsoft`?

**Comment by EgorBo** [comment]
PTAL @AndyAyersMS (I realized that I also don't know what `m` stands for exactly 😄)

**Comment by AndyAyersMS** [comment]
Not sure either. @davidwrighton is the originator of "mibc"...

File: docs/project/glossary.md (hunk 1 above)
Comment (danmoseley) [reply docs/project/glossary.md:34]: @davidwrighton ?

Comment (JulieLeeMSFT) [reply docs/project/glossary.md:34]: CC @BruceForstall.

Comment (davidwrighton) [reply docs/project/glossary.md:34]: Managed was what I was thinking, but I don't have a strong opinion here.

Comment (am11) [reply docs/project/glossary.md:34]:
```suggestion
| MIBC | Managed Instrumented Block Counts - used as extension (`*.mibc`) for managed PGO files. |
```

**Review by AndyAyersMS — APPROVED** [review]

**Pull request merged**